### PR TITLE
[Codepush] Migration of utils subpackage, refactor exceptions handling

### PR DESCRIPTION
**Pull request by ruslan-bikkinin**
Just implementation, tests will be added in the next PR.
Previous PR#13 was closed by me.

### CODE REVIEW

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/exceptions/CodePushInvalidPublicKeyException.java [hunk 1]
```diff
@@ -3,14 +3,14 @@
 /**
  * Exception class for throwing invalid CodePush public key exceptions.
  */
```
Comment (annakocheshkova) [inline sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/exceptions/CodePushInvalidPublicKeyException.java]: I'd suggest provide an extra line of info on invalid public keys.

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/exceptions/CodePushInvalidUpdateException.java [hunk 2]
```diff
@@ -3,14 +3,25 @@
 /**
  * Exception class for throwing invalid CodePush update exceptions.
  */
-public class CodePushInvalidUpdateException extends RuntimeException {
```
Comment (annakocheshkova) [inline sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/exceptions/CodePushInvalidUpdateException.java]: Here, also, I would provide info about in what cases an update is considered invalid.

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/exceptions/CodePushSignatureVerificationException.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+     * Constructs exception with with specified detail message.
+     *
+     * @param message detailed message.
+     */
+    public CodePushSignatureVerificationException(String message) {
+        super(message);
+    }
+
+    /**
+     * Constructs exception with with default detail message and specified cause.
```
Comment (annakocheshkova) [inline sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/exceptions/CodePushSignatureVerificationException.java]: with with

Comment (ruslan-bikkinin) [reply sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/exceptions/CodePushSignatureVerificationException.java]: @annakocheshkova thx, I'll fix it.

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/utils/CodePushUpdateUtils.java [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+        return expectedHash.equals(updateContentsManifestHash);
+    }
+
+    /**
+     * Verifies and decodes JWT.
+     *
+     * @param jwt       JWT string.
+     * @param publicKey public key for verification.
+     * @return <i>claims</i> value of decoded payload or null if error occurred.
+     */
```
[the remark this answers is not in the document]
Comment (ruslan-bikkinin) [reply sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/utils/CodePushUpdateUtils.java]: @MatkovIvan thx for the tip!

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/utils/CodePushUpdateUtils.java [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+    /**
+     * Parses public key from string into {@link PublicKey} class instance.
+     *
+     * @param stringPublicKey input public key value.
+     * @return parsed {@link PublicKey} class instance.
+     * @throws CodePushSignatureVerificationException if there as an error during public key parsing.
+     */
+    @SuppressWarnings("WeakerAccess")
+    public static PublicKey parsePublicKey(String stringPublicKey) throws CodePushSignatureVerificationException {
+            /* Remove unnecessary "begin/end public key" entries from string. */
```
[the remark this answers is not in the document]
Comment (ruslan-bikkinin) [reply sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/utils/CodePushUpdateUtils.java]: @MatkovIvan thx.

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/utils/CodePushUpdateUtils.java [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+                CodePushConstants.BUNDLE_JWT_FILE_NAME
+        );
+    }
+
+    /**
+     * Returns JWT content of local update.
+     *
+     * @param folderPath local update directory path.
+     * @return JWT content of update.
+     */
```
[the remark this answers is not in the document]
Comment (ruslan-bikkinin) [reply sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/utils/CodePushUpdateUtils.java]: @MatkovIvan thx, I'll fix it.

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/utils/CodePushUtils.java [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+     */
+    @SuppressWarnings("WeakerAccess")
+    public static JSONObject getJsonObjectFromFile(String filePath) throws CodePushMalformedDataException {
+        try {
+            String content = FileUtils.readFileToString(filePath);
+            return new JSONObject(content);
+        } catch (JSONException | IOException e) {
+            throw new CodePushMalformedDataException(
+                    "Unable to parse contents of \"" + filePath + "\", the file may be corrupted.",
+                    e
```
[the remark this answers is not in the document]
Comment (ruslan-bikkinin) [reply sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/utils/CodePushUtils.java]: @MatkovIvan I'll take care of it, thx!

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/exceptions/CodePushInvalidPublicKeyException.java (hunk 1 above)
Comment (ruslan-bikkinin) [reply sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/exceptions/CodePushInvalidPublicKeyException.java]: @annakocheshkova thx, i'll update the info.

**Pull request merged**